OktaIdxExample/OktaIdxExampleUITests/Utilities/Scenario.swift

```diff
@@ -102,18 +102,6 @@ struct Scenario {
         
         var errors = [Swift.Error]()
         XCTContext.runActivity(named: "Tearing down test configuration") { _ in
-            if let credentials = credentials {
```

Comment:
Why are you not deleting the user at the end of the test?

Reply:
To clarify, this class is used in _all_ E2E tests, so skipping the delete here will impact the other tests.

Also, this `tearDown` method is only run at the end of _all_ tests within a particular test class, so your user accounts should exist between test methods, unless you need to explicitly delete them between tests, at which point it can be done in the test class' tearDown

Reply:
There are the reasons of that:
1. > unless you need to explicitly delete them between tests, at which point it can be done in the test class' tearDown

Scenarios 6.2.2 and 6.2.4 have the prerequisite - user must be already enrolled. This forces me to delete the user between tests. When the tests are finished (as you said) - `tearDown` method is run at the end of all tests. So the user doesn't exists, tests are failed due to `tearDown` fail.

2. We [explicitly create a user](https://github.com/okta/okta-idx-swift/blob/78b3cbc969b0f6b5417d34d8f901a60f89622c5a/OktaIdxExample/OktaIdxExampleUITests/PhoneLoginScenarioTests.swift#L22), so I would suggest explicitly delete it in object method `tearDown`. We can move this logic to `ScenarioTessCase` if it repeats for major vast of cases.

3. Enrolling of user and authentication are slightly different. I ended up with dull code `ifs` when I tried to generalize the code. So in this scenario I think deleting and creating the user between tests, enrolling it before testing specific scenario is a tradeoff. Moreover, order of tests are run random, so we hardly know whether a user is enrolled already or not, so we should it [detect somehow](https://github.com/okta/okta-idx-swift/pull/38/files#diff-70b314b4b0988ed5a5add42c3f3b651396b5390be0fb8129b418e6703dc6d9a2R22) and put [`ifs`](https://github.com/okta/okta-idx-swift/pull/38/files#diff-db750673b977f8f5406987b9269814e327b3ecb76b1e24a25e5f9f78a737c00cR107). 
It makes tests hard to debug.

Reply:
Yes, but 6.2.2 and 6.2.4 are within the same test class, so the user won't be deleted between test runs. 

If you'll notice, the scenario tearDown method is called when the test class's tearDown method is call, not the test method.

```
override class func tearDown()
```

According to the [Apple docs on setUp and tearDown](https://developer.apple.com/documentation/xctest/xctestcase/understanding_setup_and_teardown_for_test_methods), it says that the setUp and tearDown methods are called before and after _all_ tests within that class are run.  The instance method versions of setUp and tearDown are called between tests within that same class.

So the user shouldn't be deleted until all tests within this class are run.

Reply:
But I do need delete between test runs. Because scenarios 6.2.2 and 6.2.4 require a user enrolled. 

Ok, I'll refactor the scenario the test to revert the changes.

Reply:
@oleggnidets-okta In that case, you can just make extra calls to deleteUser() in the instance-method `tearDown` for just that one class.